apps/science/src/main.cpp

```diff
@@ -1,19 +1,143 @@
-#include "CANBus.h"
+#include "ActuatorController.h"
+#include "CANBuffer.h"
 #include "CANMsg.h"
+#include "Encoder.h"
+#include "EncoderAbsolute_PWM.h"
+#include "EncoderRelative_Quadrature.h"
+#include "LimServo.h"
+#include "MoistureSensor.h"
+#include "ScienceConfig.h"
 #include "hw_bridge.h"
 #include "mbed.h"
 
+// motors
+Motor elevatorMotor(MTR_PWM_2, MTR_DIR_2, false);
+Motor indexerMotor(MTR_PWM_1, MTR_DIR_1, false);
+
+// Servo
+LimServo coverServo(SRVO_PWM_1, ScienceConfig::coverServoRange, ScienceConfig::coverServoMaxPulse,
+                    ScienceConfig::coverServoMinPulse);
+
+LimServo diggerServo(SRVO_PWM_2, ScienceConfig::diggerServoRange, ScienceConfig::diggerServoMaxPulse,
+                     ScienceConfig::diggerServoMinPulse);
+
+// encoders
+EncoderAbsolute_PWM elevatorEncoder(ScienceConfig::elevatorEncoderConfig);
+EncoderRelative_Quadrature indexerEncoder(ScienceConfig::centrifugeEncoderConfig);
+
+// limit switches
+DigitalIn elevatorLimTop(LIM_SW_3);
+DigitalIn elevatorLimBottom(LIM_SW_4);
+DigitalIn indexerLimFront(LIM_SW_1);
+DigitalIn indexerLimBack(LIM_SW_2);
+
+// Actuators
+ActuatorController elevatorActuator(ScienceConfig::diggerLiftActuatorConfig, elevatorMotor, elevatorEncoder,
+                                    elevatorLimBottom, elevatorLimTop);
+ActuatorController indexerActuator(ScienceConfig::indexerActuatorConfig, indexerMotor, indexerEncoder, indexerLimFront,
+                                   indexerLimBack);
+
+// I2C
+MoistureSensor moistureSensor = MoistureSensor(TEMP_MOIST_I2C_SDA, TEMP_MOIST_I2C_SCL);
+
+// LEDs
 DigitalOut led1(LED1);
+DigitalOut ledR(LED_R);
+DigitalOut ledG(LED_G);
+DigitalOut ledB(LED_B);
 
-constexpr auto k_interval = 500ms;
+// Declaring can object
+CAN can(CAN1_RX, CAN1_TX, HWBRIDGE::ROVERCONFIG::ROVER_CANBUS_FREQUENCY);
```

Comment:
CAN class is deperecated. use CANBus class. its basically the same

Reply:
done